[WIP] REACT Completed 4, task 5 in progress

NikitaKhadnevich opened the pull request.

k-krasilnikova commented on React_Notes_Block/src/components/notes/ActionButtons.jsx:
```diff
@@ -0,0 +1,93 @@
+import PropTypes from 'prop-types';
+import Stack from '@mui/material/Stack';
+
+const ButtonChange = (props) => {
```
Pretty strange name of the component. Let's rename it to something more clear(Example: `EditButton`)

NikitaKhadnevich replied:
Renamed this Button-elem

k-krasilnikova commented on React_Notes_Block/src/components/notes/ActionButtons.jsx:
```diff
@@ -0,0 +1,93 @@
+import PropTypes from 'prop-types';
+import Stack from '@mui/material/Stack';
+
+const ButtonChange = (props) => {
+  const { id, changeNote, callToEditNote, noteList } = props;
```
There is no need to define an additional constants at the new line:
`const ButtonChange = ({ id, changeNote, callToEditNote, noteList }) => {`

NikitaKhadnevich replied:
Deleted const and made it into func-params

k-krasilnikova commented on React_Notes_Block/src/components/notes/ActionButtons.jsx:
```diff
@@ -0,0 +1,93 @@
+  const { id, changeNote, callToEditNote, noteList } = props;
+
+  return (
+    <>
+      <IconButton
+        aria-label='edit'
+        onClick={() => changeNote(id, noteList, callToEditNote)}
+      >
+        <ModeEditIcon />
+      </IconButton>
+    </>
```
No need to wrap `IconButton` with an additional empty tag

NikitaKhadnevich replied:
Deleted empty tag

k-krasilnikova commented on React_Notes_Block/src/components/notes/ActionButtons.jsx:
```diff
@@ -0,0 +1,93 @@
+
+export default ButtonChange;
+
+export const ButtonSaveChanges = (props) => {
```
Only one component per file!
You can move all the actions to a separate files: `components/notes/ActionButtons/SaveButton/SaveButton.jsx`
`components/notes/ActionButtons/EditButton/EditButton.jsx`
`components/notes/ActionButtons/RemoveButton/RemoveButton.jsx`
And compose them inside `components/notes/ActionButtons/ActionButtons.jsx`

NikitaKhadnevich replied:
Added new files and replace its

k-krasilnikova commented on React_Notes_Block/src/components/notes/ActionButtons.jsx:
```diff
@@ -0,0 +1,93 @@
+
+export default ButtonChange;
+
+export const ButtonSaveChanges = (props) => {
```
Define all the props inside params by destructing it

NikitaKhadnevich replied:
Done

k-krasilnikova commented on React_Notes_Block/src/components/notes/ActionButtons.jsx:
```diff
@@ -0,0 +1,93 @@
+export const ButtonSaveChanges = (props) => {
+  const { id, saveEditedNote, newInputState } = props;
+  return (
+    <>
```
Remove please empty tag

NikitaKhadnevich replied:
Done

k-krasilnikova commented on React_Notes_Block/src/components/notes/ActionButtons.jsx:
```diff
@@ -0,0 +1,93 @@
+        <IconButton
+          aria-label='delete'
+          color='primary'
+          onClick={() => deleteNotes(index)}
```
`deleteNote`, only one..

NikitaKhadnevich replied:
Renamed to deleteNote

k-krasilnikova commented on React_Notes_Block/src/components/notes/InputChange.jsx:
```diff
@@ -0,0 +1,77 @@
+import useStyles from '../../pages/mainNotes/listNotes/styled';
+import { ButtonSaveChanges, ButtonDelete } from './ActionButtons';
+
+const InputChange = ({
```
Separate folder for each component please!

NikitaKhadnevich replied:
Separated and importing comp form new folders/files

k-krasilnikova commented on React_Notes_Block/src/components/notes/InputChange.jsx:
```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react';
+import Box from '@mui/material/Box';
+import TextField from '@mui/material/TextField';
+import PropTypes from 'prop-types';
+import { Slide } from '@material-ui/core';
+import useStyles from '../../pages/mainNotes/listNotes/styled';
+import { ButtonSaveChanges, ButtonDelete } from './ActionButtons';
```
If we're talking about imports, they separated by 3 categories:
```
libraries

absolute paths

relative paths
```
Example:
```
import React, { useState } from "react";
import Box from "@mui/material/Box";

import useStyles from 'pages/mainNotes/listNotes/styled';

import { ButtonSaveChanges, ButtonDelete } from './ActionButtons';
```
1. Setup absolute paths for pages, config, components folders
2. Sort imports by category

NikitaKhadnevich replied:
Fixed in a few files, but i recognized that 'Eslin' make it automatically

k-krasilnikova commented on React_Notes_Block/src/components/notes/InputChange.jsx:
```diff
@@ -0,0 +1,77 @@
+  currentState,
+}) => {
+  const [newInputState, setNewInputState] = useState(description);
+  const classes = useStyles();
```
Import please from `styled` file already prepared classes, or wrap component with MUi `useStyles` to avoid getting classes in separate constant

NikitaKhadnevich replied:
There were destructure styles and fixed classNames.

k-krasilnikova commented on React_Notes_Block/src/components/notes/InputChange.jsx:
```diff
@@ -0,0 +1,77 @@
+  index,
+  currentState,
+}) => {
+  const [newInputState, setNewInputState] = useState(description);
```
Is there any cases for setting old state?
No need to write new, and also inout, here we can set 100000 inputs and what will be the differences?
```suggestion
  const [updatedDescription, setDescription] = useState(description);
```

NikitaKhadnevich replied:
Renamed its

k-krasilnikova commented on React_Notes_Block/src/components/notes/InputChange.jsx:
```diff
@@ -0,0 +1,77 @@
+  const [newInputState, setNewInputState] = useState(description);
+  const classes = useStyles();
+
+  const sendNewValue = (e) => {
```
old value?
what is value?
Rename please to something more clear

NikitaKhadnevich replied:
Fixed its also

k-krasilnikova commented on React_Notes_Block/src/components/notes/InputChange.jsx:
```diff
@@ -0,0 +1,77 @@
+    <Slide in={isChange}>
+      <Box
+        className={isChange ? classes.changeInputShow : classes.changeInputHide}
+        style={{ marginLeft: '0px' }}
```
Move all the styles to a separate file

k-krasilnikova commented on React_Notes_Block/src/pages/header/HeaderNav.jsx:
```diff
@@ -0,0 +1,19 @@
+const HeaderNav = () => {
+  const classes = useStyles();
+  return (
+    <>
```
!

NikitaKhadnevich replied:
Done

k-krasilnikova commented on React_Notes_Block/src/pages/header/HeaderNav.jsx:
```diff
@@ -0,0 +1,19 @@
+import useStyles from './styled';
+
+const HeaderNav = () => {
+  const classes = useStyles();
```
!

NikitaKhadnevich replied:
Fixed

k-krasilnikova commented on React_Notes_Block/src/pages/header/HeaderNavCont.jsx:
```diff
@@ -0,0 +1,10 @@
+import HeaderNav from './HeaderNav';
+
+const HeaderNavCont = () => (
+  <>
```
!

NikitaKhadnevich replied:
Done

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/aboutNote/AboutNote.jsx:
```diff
@@ -0,0 +1,41 @@
+  const classes = useStyles();
+
+  const addDescription = () => {
+    const sortDescription = mainNotes.filter((item) => item.isActive);
```
I'm not sure that it's a good practice to save result in array, try to get directly selected object.

NikitaKhadnevich replied:
destructured to object and replaced 'arr[0]....'  into 'if.. else'

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/aboutNote/AboutNote.jsx:
```diff
@@ -0,0 +1,41 @@
+  }, [mainNotes]);
+
+  return (
+    <>
```
!

NikitaKhadnevich replied:
done

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/listNotes/ListNotes.jsx:
```diff
@@ -0,0 +1,134 @@
+
+  const handleItem = (id, currentState = noteList, actionFunction) => {
+    const filtredNotes = actionFunction(id, currentState);
+    setNoteList(() => [...filtredNotes]);
```
Why did you use arrow function inside `setNoteList`? Why did you add rest prop?
```suggestion
    setNoteList(filtredNotes);
```

NikitaKhadnevich replied:
I thought that would use 'prev => ...prev' , but its isnt my case, after some time i refused it and.... forgot to fix

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/listNotes/ListNotes.jsx:
```diff
@@ -0,0 +1,134 @@
+  };
+  const handleDelete = (id, currentState = noteList) => {
+    const actualNotes = deleteNote(id, currentState);
+    setNoteList(() => [...actualNotes]);
```
?

NikitaKhadnevich replied:
Fixed

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/listNotes/ListNotes.jsx:
```diff
@@ -0,0 +1,134 @@
+  };
+  const handleSaveNote = (id, valueFromInput) => {
+    const savedNotes = saveEditedNote(id, noteList, valueFromInput);
+    setNoteList(() => [...savedNotes]);
```
?

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/listNotes/ListNotes.jsx:
```diff
@@ -0,0 +1,134 @@
+      : setToLocalStorage(noteList);
+  }, [noteList]);
+  return (
+    <>
```
!

NikitaKhadnevich replied:
Replaced

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/listNotes/ListNotes.jsx:
```diff
@@ -0,0 +1,134 @@
+                        variant='body2'
+                        className={classes.description}
+                      >
+                        {item.description.slice(0, SHORT_DESC.limit)}
```
Move it please to the `utils/index.js` and call here just function for cutting content

NikitaKhadnevich replied:
Moved into separate file

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/listNotes/ListNotes.jsx:
```diff
@@ -0,0 +1,134 @@
+                  </Box>
+
+                  <InputChange
+                    key={Math.random()}
```
I'm not sure it's a good idea

NikitaKhadnevich replied:
made 'custom' keys , which based on item.id

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/mainNotesCont.jsx:
```diff
@@ -0,0 +1,27 @@
+import React, { useState, Suspense } from 'react';
+import useStyles from './styled';
+
+const ListNotes = React.lazy(() => import('./listNotes/ListNotes'));
+const AboutNote = React.lazy(() => import('./aboutNote/AboutNote'));
```
We can create index.jsx files inside mainNotes/ListNotes and MainNotes/AboutNotes folders that will contain lazy loading. So there is no need to overload container with such types of imports

NikitaKhadnevich replied:
Added new jsx files and put Lazy

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/mainNotesCont.jsx:
```diff
@@ -0,0 +1,27 @@
+import React, { useState, Suspense } from 'react';
```
What is mainNotesCont? Container? Contributor?

NikitaKhadnevich replied:
its Container

k-krasilnikova replied:
Can't see any changes with name

NikitaKhadnevich replied:
Fixed

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/mainNotesCont.jsx:
```diff
@@ -0,0 +1,27 @@
+  };
+
+  return (
+    <>
```
!

NikitaKhadnevich replied:
fixed again

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/mainNotesCont.jsx:
```diff
@@ -0,0 +1,27 @@
+
+  return (
+    <>
+      <article className={classes.mainContainer}>
+        <Suspense fallback={<div className={classes.spiner}>Loading...</div>}>
+          <AboutNote mainNotes={mainNotes} />
+          <ListNotes sentAbout={sentAbout} />
+        </Suspense>
+      </article>
```
You should define it's container or component.... logic in container, UI(only) inside the component

NikitaKhadnevich replied:
Its Container ,fixed

k-krasilnikova commented on React_Notes_Block/src/utils/CallToEditNote.jsx:
```diff
@@ -0,0 +1,14 @@
+const callToEditNote = (buttonId, matchSource) => {
+  let arrNoteReciever = [...matchSource];
```
Why rest again?

NikitaKhadnevich replied:
Fixed, I thing its is not bind with prev => ..prev, Its realy my mistake as like in Handler-func (

k-krasilnikova commented on React_Notes_Block/src/utils/ChooseNote.jsx:
```diff
@@ -0,0 +1,14 @@
+const chooseNote = (itemId, matchSource) => {
+  let arrNoteReciever = [...matchSource];
```
?

NikitaKhadnevich replied:
Fixed

k-krasilnikova commented on React_Notes_Block/src/utils/ChooseNote.jsx:
```diff
@@ -0,0 +1,14 @@
+const chooseNote = (itemId, matchSource) => {
```
No need to create for each util new file, we can define all of them in one:
```
export const choose note ...
export const callEditNote ...
```

NikitaKhadnevich replied:
Yes, these are similar function, but each of them change different params = callToEdit works with  **...note, isActive: !note.isActive, isChange: !note.isChange ,**
chooseNote works with  **...note, isActive: !note.isActive, isChange: false** . There is important for styles and note appearance. Maybe will remain these function?

k-krasilnikova replied:
Just move all of them inside one file

NikitaKhadnevich replied:
Moved inside ChooseNote.jsx

k-krasilnikova commented on React_Notes_Block/src/utils/ChooseNote.jsx:
```diff
@@ -0,0 +1,14 @@
+const chooseNote = (itemId, matchSource) => {
+  let arrNoteReciever = [...matchSource];
+  const toggleItem = (arr, id) => {
```
arr?

NikitaKhadnevich replied:
renamed

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/listNotes/ListNotes.jsx:
```diff
@@ -0,0 +1,134 @@
+  }, [noteList]);
+  return (
+    <Grid container spacing={2} className={gridNotes}>
+      {noteList
+        ? noteList.map((item, index) => (
```
We can use here optional chaining.
```suggestion
      {noteList?.map((item, index) => (
```

k-krasilnikova replied:
Or `noteList && noteList.map..`
It's not a good practice to use `null` inside jsx

NikitaKhadnevich replied:
```suggestion
      {noteList
        ? noteList.map((item, index) => (
```
Remake notelist and added other component instead null

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/mainNotesCont.jsx:
```diff
@@ -0,0 +1,25 @@
+import LazyAboutNotes from './aboutNote/LazyAboutNotes';
+
+const MainNotesCont = () => {
+  const { mainContainer, spiner } = useStyles();
```
remove

NikitaKhadnevich replied:
Done

NikitaKhadnevich commented on React_Notes_Block/src/components/notes/InputChange.jsx:
```diff
@@ -0,0 +1,75 @@
+import Box from '@mui/material/Box';
+import TextField from '@mui/material/TextField';
+import { Slide } from '@material-ui/core';
+import useStyles from '../../pages/mainNotes/listNotes/styled';
```
On one with folder level, added wrapper-reciever for this.

NikitaKhadnevich commented on React_Notes_Block/src/pages/header/HeaderNav.jsx:
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { AppBar, Toolbar } from '@material-ui/core';
+import IconName from '../../components/header/IconName';
+import useStyles from './styled';
```
Overwrote styles

NikitaKhadnevich commented on React_Notes_Block/src/pages/header/HeaderNavCont.jsx:
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+import HeaderNav from './HeaderNav';
+
+const HeaderNavCont = () => <HeaderNav />;
```
Deleted this item

NikitaKhadnevich commented on React_Notes_Block/src/pages/mainNotes/aboutNote/AboutNote.jsx:
```diff
@@ -0,0 +1,40 @@
+import PropTypes from 'prop-types';
+import { Typography, Container } from '@material-ui/core';
+import useStyles from './styled';
+import { INITIALNOTE } from '../../../config/constants/initNoteData';
```
Make it

NikitaKhadnevich commented on React_Notes_Block/src/pages/mainNotes/aboutNote/AboutNote.jsx:
```diff
@@ -0,0 +1,40 @@
+
+const AboutNote = ({ mainNotes }) => {
+  const [aboutData, setAboutData] = useState();
+  const { aboutWrapper } = useStyles();
```
Done

NikitaKhadnevich commented on React_Notes_Block/src/pages/mainNotes/listNotes/ListNotes.jsx:
```diff
@@ -0,0 +1,134 @@
+
+import useStyles from './styled';
+
+import WAHA_NOTES from '../../../config/constants/initNoteData';
+import chooseNote from '../../../utils/ChooseNote';
+import callToEditNote from '../../../utils/CallToEditNote';
+import saveEditedNote from '../../../utils/SaveEditedNote';
+import sliceDescription from '../../../utils/SliceDescription';
+import deleteNote from '../../../utils/DeleteNote';
+import ButtonEdit from '../../../components/notes/actionsButtons/ButtonEdit';
+import InputChange from '../../../components/notes/InputChange';
```
Fixed

NikitaKhadnevich commented on React_Notes_Block/src/pages/mainNotes/listNotes/ListNotes.jsx:
```diff
@@ -0,0 +1,134 @@
+import useStyles from './styled';
+
+import WAHA_NOTES from '../../../config/constants/initNoteData';
+import chooseNote from '../../../utils/ChooseNote';
+import callToEditNote from '../../../utils/CallToEditNote';
+import saveEditedNote from '../../../utils/SaveEditedNote';
+import sliceDescription from '../../../utils/SliceDescription';
+import deleteNote from '../../../utils/DeleteNote';
```
Done

NikitaKhadnevich commented on React_Notes_Block/src/pages/mainNotes/listNotes/ListNotes.jsx:
```diff
@@ -0,0 +1,134 @@
+    title,
+    description,
+    noteActions,
+  } = useStyles();
```
Done

NikitaKhadnevich commented on React_Notes_Block/src/pages/mainNotes/listNotes/ListNotes.jsx:
```diff
@@ -0,0 +1,134 @@
+      ? setToLocalStorage(WAHA_NOTES)
+      : setToLocalStorage(noteList);
+  }, [noteList]);
+  return (
```
Separated this block, now its Container/Component

NikitaKhadnevich commented:
In this request I had done:

1. Added some func and styles
2. Separate Cont/Comp;
3. Created 'Wrapper-Recivers' for path to files;
4. Added Spinner for loading;
5. Fixed func and method;
6. Added constant;
7. Overwrote useStyles to Styles Comp;
8. Replace modules and folders

k-krasilnikova commented on React_Notes_Block/src/components/forms/SignUp.jsx:
```diff
@@ -0,0 +1,185 @@
+    handleSubmit,
+  } = formik;
+
+  function change(name, e) {
```
Why it's not an arrow?

NikitaKhadnevich replied:
Renamed and converted to arrow

k-krasilnikova commented on React_Notes_Block/src/components/forms/SignUp.jsx:
```diff
@@ -0,0 +1,185 @@
+
+SignUp.propTypes = {
+  // email: PropTypes.string,
+  // confirmPassword: PropTypes.string,
```
Do we really need this commented code?

NikitaKhadnevich replied:
Yes, I will use this prop in future. There are formik values, i will implemented or remove in next pull. Its ok? May i save this?

k-krasilnikova replied:
Yep, sure

NikitaKhadnevich replied:
Implement it in this push

k-krasilnikova commented on React_Notes_Block/src/pages/aboutPage/LazyAboutPage.jsx:
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
```
We should rename this file to something more simple:
index.jsx

NikitaKhadnevich replied:
fixed to indexLazy.jsx

k-krasilnikova commented on React_Notes_Block/src/pages/aboutPage/LazyAboutPage.jsx:
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
+const LazyAboutPage = React.lazy(() => import('./aboutPage'));
```
`aboutPage` inside `aboutPage`? Rewrite this structure please.

NikitaKhadnevich replied:
Yes, Lets see. I configured this files as like in article which u sent.
I have container with some html, and i want accurate this container be lazy.
For this i sent it to React Lazy and then wrapped In suspense.
For example in future i will want to add some component without lazy-load and spinner.  Its will be easy - i will add it in Main container. In the end i will have 2 components: with lazy and without. I think its good idea. about note block in "notes" work similar. As user i wdnt see 'spinner hell'))

NikitaKhadnevich replied:
Fixed to index.js - style

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/aboutNote/AboutNote.jsx:
```diff
@@ -0,0 +1,34 @@
+  const [aboutData, setAboutData] = useState();
+
+  const addDescription = () => {
+    const [sortDescription] = mainNotes.filter((item) => item.isActive);
```
What is `item`?

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/aboutNote/AboutNote.jsx:
```diff
@@ -0,0 +1,34 @@
+    addDescription();
+  }, [mainNotes]);
+
+  return <AboutNoteCont aboutData={aboutData} />;
```
This file should be renamed as `AboutNoteContainer`, and `AboutNoteCont` -> `AboutNote`
Read please rules again. `Cont` is pretty bad word in naming

NikitaKhadnevich replied:
Renamed and replace path to 'App.jsx'

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/aboutNote/LazyAboutNotes.jsx:
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
```
index!

NikitaKhadnevich replied:
Remove lazy from this block, Its small elem and i suppose its not good idea wrapping everything in lazy

NikitaKhadnevich replied:
Fixed to index.js - style

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/listNotes/LazyListNotes.jsx:
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
```
index

NikitaKhadnevich replied:
renamed to indexLazy.jsx

NikitaKhadnevich replied:
Fixed to index.js - style

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/listNotes/ListNotesCont.jsx:
```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Typography, Grid } from '@material-ui/core';
+import ListItem from '@mui/material/ListItem';
```
It's a component...
Remove `Cont` from all the places inside the application. Use `Container` instead, but in correct places!!!!!!!

NikitaKhadnevich replied:
Fixed and renamed

k-krasilnikova commented on React_Notes_Block/src/pages/mainNotes/mainNotes.jsx:
```diff
@@ -0,0 +1,96 @@
+  removeShareNote,
+} from './mainNotesReceiver';
+
+const MainNotes = () => {
```
It's a Container

NikitaKhadnevich replied:
Replaced

k-krasilnikova commented on React_Notes_Block/src/pages/sharedNotePage/sharedNotesCont.jsx:
```diff
@@ -0,0 +1,63 @@
+  NoSharedNotes,
+} from './styled';
+
+const SharedNotes = () => {
```
It's a Component

NikitaKhadnevich replied:
Fixed and renamed

k-krasilnikova reviewed (changes requested):
Good job, but still needs some updates

NikitaKhadnevich commented:
Finished work with 4 Task. In this Pull i finally had fixed wrongs in files and folder and added SignUp/SignIn components, Also added new Redux actions/reducer for this and other. Also I worked with React Query (query, mutaions) and LocalStorage.

k-krasilnikova commented on React_Notes_Block/src/api/sharedToolkit/sharedSelectors.js:
```diff
@@ -0,0 +1,3 @@
+export const sharedArr = (state) => state.notesShare.sharedArr;
```
arr, mrr, prr, brr.
Rename this variable with something more clear please

NikitaKhadnevich replied:
Renamed to sharedNotes

k-krasilnikova commented on React_Notes_Block/src/api/sharedToolkit/sharedSelectors.js:
```diff
@@ -0,0 +1,3 @@
+export const sharedArr = (state) => state.notesShare.sharedArr;
+export const isShared = (state) => state.inotesShare.sShared;
```
seems typo here

NikitaKhadnevich replied:
yep , fixed

k-krasilnikova commented on React_Notes_Block/src/components/forms/FormikWrappers/FormikIn.js:
```diff
@@ -0,0 +1,73 @@
+          querryData={data}
+        />
+      </Container>
+    </>
```
Remove unexpected wrapper please

NikitaKhadnevich replied:
Removed

k-krasilnikova commented on React_Notes_Block/src/components/forms/FormikWrappers/FormikUp.js:
```diff
@@ -0,0 +1,77 @@
+  });
+
+  return (
+    <>
```
Unexpected wrapper
Also please rename this file to the: FormikUpContainer

NikitaKhadnevich replied:
Renamed this and FormikIn => FormikUpContainer

k-krasilnikova commented on React_Notes_Block/src/components/forms/SignIn.jsx:
```diff
@@ -0,0 +1,154 @@
+  };
+
+  return (
+    <SignContainer component='main'>
```
Is it container? Or component? Separate logic from UI and create an appropriate name for them

NikitaKhadnevich replied:
This is component, and logic had been separated before.I added this name only in styled to Material UI Component. Previously it was 'Material UI Container' and I added name which based on main elem-name (Container). For example, i see that Material UI Style name is SignContainer. This name include the  string 'Container' and i understand that  'Material UI' type is Container. Sometimes its help to identify component style without look up into styled.jsx.
I see that is not good idea and i renamed this and other names to 'Wrapper' stuff like that.

k-krasilnikova commented on React_Notes_Block/src/components/forms/SignUp.jsx:
```diff
@@ -0,0 +1,220 @@
+  };
+
+  return (
+    <SignContainer component='main'>
```
Container? Component?

NikitaKhadnevich replied:
Renamed to Wrapper

NikitaKhadnevich commented:
**_These commit include a lot of changes:_**
-	Fixed names and paths a few files;
-	Split into small large components as like _ListNotes_, _SignUp_, _SignIn_ etc.;
-	Renamed variables in some components;
-	Rewrote ‘_NoteList State_’ into ‘_NoteList fake API_’ (added new data-base for receive, save and syncing , added useQuery, useMutations also created RestAPI functions for work with this;
-	Created Input for fill new Note and created ‘Formik-logic’ for work with this;
-	Added Box-item and functions for sorting current notes;
-	Fixed some bugs, replace styles;

k-krasilnikova commented on React_Notes_Block/src/api/RESTClient/apiConstants.jsx:
```diff
@@ -0,0 +1,18 @@
+const baseURL = 'http://localhost:3001/users';
+const notesURL = 'http://localhost:3100/WAHA_NOTES';
```
Uppercase please!

NikitaKhadnevich replied:
Added Upper

k-krasilnikova commented on React_Notes_Block/src/components/forms/FormikWrappers/FormikInContainer.js:
```diff
@@ -0,0 +1,71 @@
+  const MIN_ARR_LENGTH = 1;
+  const navToNotes = useNavigate();
+
+  const { isLoading, error, data, isFetching } = useQuery(
+    'GetUsersForAuth',
+    () => fetch(baseURL).then((responseUsers) => responseUsers.json())
+  );
```
Move please `useQuery(` to the `src/api/auth` folder. Write your own hook ("wrapper for useQuery"  `getUsers`) and call it inside Container

NikitaKhadnevich replied:
Created new Folder and ownHooks, but one of them stayed old, its contain 'onSuccess logical' (ForrmikUp) block

k-krasilnikova commented on React_Notes_Block/src/components/forms/FormikWrappers/FormikUpContainer.js:
```diff
@@ -0,0 +1,74 @@
+  const { hasAlready } = ERROR_MESSAGES;
+  const MIN_ARR_LENGTH = 1;
+
+  const { isLoading, error, data, isFetching } = useQuery('GetUsers', () =>
```
!

NikitaKhadnevich replied:
Done

k-krasilnikova commented on React_Notes_Block/src/components/forms/SignInputs/SIgnUpBirth.jsx:
```diff
@@ -0,0 +1,47 @@
+    helperText={
+      touched.birthday && birthday.length > MIN_LENGTH_BIRTH
+        ? errors.birthday
+        : ''
```
It's not a good practice to use empty strings, try to replace condition with `&&`

NikitaKhadnevich replied:
Fixed

k-krasilnikova reviewed (changes requested):
Update please and merge the PR

NikitaKhadnevich commented on React_Notes_Block/src/api/RESTClient/RESTApi.js:
```diff
@@ -0,0 +1,35 @@
+    .catch((err) => new Error(getError));
+}
+
+function runPOSTuser(baseUrl, obj) {
```
Fixed
